src/services/openai.ts

```diff
@@ -84,67 +89,100 @@ export class OpenAIService {
+          deckConfig.prompt,
+          card,
+          deckConfig,
+        );
+
+        const fullPrompt = `${processedPrompt}\n\n${this.generateResponseFormat(deckConfig)}`;
+        this.debugLog(fullPrompt);
+        this.debugLog(this.chatModel);
+        const completion = await this.client.chat.completions.create({
+          model: this.chatModel,
+          messages: [{ role: "user", content: fullPrompt }],
+          response_format: { type: "json_object" },
+        });
+        
+        const response = JSON.parse(
+          // biome-ignore lint/style/noNonNullAssertion: <explanation>
+          completion.choices[0].message.content!,
+        ) as Record<string, string>;
 
-      const response = JSON.parse(
-        // biome-ignore lint/style/noNonNullAssertion: <explanation>
-        completion.choices[0].message.content!,
-      ) as Record<string, string>;
 
-      // Validate required fields
-      const missingRequired = Object.entries(deckConfig.responseFields)
-        .filter(([field, config]) => config.required && !response[field])
-        .map(([field]) => field);
+        // Validate required fields
+        const missingRequired = Object.entries(deckConfig.responseFields)
+          .filter(([field, config]) => config.required && !response[field])
+          .map(([field]) => field);
 
-      if (missingRequired.length > 0) {
-        throw new PromptError("AI response missing required fields", {
+        if (missingRequired.length === 0) {
+          return response;
+        }
+
+        this.debugLog("AI response missing required fields", {
           missingFields: missingRequired,
         });
+        this.debugLog("Now Regenerating");
+
+      } catch (error) {
+        if (error instanceof PromptError) {
+          throw error;
+        }
+        throw new PromptError("Content generation failed", {
+          configIssues: [(error as Error).message],
+        });
       }
-
-      return response;
-    } catch (error) {
-      if (error instanceof PromptError) {
-        throw error;
-      }
-      throw new PromptError("Content generation failed", {
-        configIssues: [(error as Error).message],
-      });
     }
   }
 
   async generateAudio(
     text: string,
     outputPath: string,
     voice: string,
+    locale = 'en-US',
   ): Promise<string> {
+    // If the model is llama, use our custom tts
+    if (this.apiKey==="ollama") {
+      try {
+        this.debugLog("Generating audio for ollama model");
+        // Use EdgeSpeechTTS for ollama model
+        const tts = new EdgeSpeechTTS({ locale });
+        const payload = {
+        input: text,
+        options: {
+          voice: voice,
+        },
+      };
+
+      const response = await tts.create(payload);
+        const mp3Buffer = Buffer.from(await response.arrayBuffer());
+        writeFile(outputPath, mp3Buffer);
+        return outputPath;
+      } catch (error) {
+        throw new PromptError("Audio generation failed", {
```

Comment:
Not a prompt error. Can be something like AudioGenerationError

Reply:
add an `AudioGenerationError` for errors generated in the audio generation process
https://github.com/Yukaii/gakuon/pull/107/commits/6df39fb8ab145854f64b5dfb3577a59a2ffcf1f2